// @govuk-react/section-break/es/index.js
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { spacing } from '@govuk-react/lib';
import { BORDER_COLOUR } from 'govuk-colours';
var breakSizes = {
  XL: 8,
  XLARGE: 8,
  L: 6,
  LARGE: 6,
  M: 4,
  MEDIUM: 4
};
var SectionBreak = styled('hr')({
  margin: 0,
  border: 0
}, function (_ref) {
  var level = _ref.level;
  var size = breakSizes[level];

  if (size) {
    return spacing.withWhiteSpace({
      margin: {
        size: size,
        direction: ['top', 'bottom']
      }
    });
  }

  return spacing.withWhiteSpace();
}, function (_ref2) {
  var visible = _ref2.visible;
  return visible ? {
    borderBottom: "1px solid " + BORDER_COLOUR
  } : undefined;
});
/**
 *
 * ### Usage
 *
 * Simple
 * ```jsx
 * <SectionBreak>Example</SectionBreak>
 * ```
 *
 * ### References
 * - https://design-system.service.gov.uk/styles/typography/#section-break
 * - https://github.com/alphagov/govuk-frontend/blob/master/src/core/_section-break.scss
 */

var SectionBreakDocumented = function SectionBreakDocumented(props) {
  return React.createElement(SectionBreak, props);
};

SectionBreakDocumented.defaultProps = {
  visible: undefined
};
SectionBreak.propTypes = SectionBreakDocumented.propTypes;
SectionBreak.defaultProps = SectionBreakDocumented.defaultProps;
export { SectionBreakDocumented };
export default SectionBreak;
