// @govuk-react/date-field/es/index.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { ERROR_COLOUR } from 'govuk-colours';
import { SPACING } from '@govuk-react/constants';
import LabelText from '@govuk-react/label-text';
import ErrorText from '@govuk-react/error-text';
import HintText from '@govuk-react/hint-text';
import { spacing } from '@govuk-react/lib';
import Input from './input';
var StyledContainer = styled('div')({
  display: 'flex',
  flexDirection: 'column'
}, function (_ref) {
  var errorText = _ref.errorText;
  return errorText ? {
    borderLeft: "4px solid " + ERROR_COLOUR,
    marginRight: SPACING.SCALE_3,
    paddingLeft: SPACING.SCALE_2
  } : undefined;
}, spacing.withWhiteSpace({
  marginBottom: 6
}));
/**
 *
 * ### Usage
 *
 * Simple
 * ```jsx
 * <DateField>What is your date of birth?</DateField>
 * ```
 *
 * Date with hint text
 * ```jsx
 * <DateField hintText="For example, 31 03 1980">
 *   What is your date of birth?
 * </DateField>
 * ```
 *
 * Date with hint text & error
 * ```jsx
 * <DateField
 *   hintText="For example, 31 03 1980"
 *   errorText="Error message goes here"
 * >
 *   What is your date of birth?
 * </DateField>
 * ```
 *
 * With custom input name props
 * ```jsx
 * <DateField hintText="For example, 31 03 1980"
 *   inputNames={{
 *     day: 'dayInputName',
 *     month: 'monthInputName',
 *     year: 'yearInputName',
 *   }}
 *  >
 *   What is your date of birth?
 * </DateField>
 * ```
 *
 * ### References:
 * - https://github.com/alphagov/govuk-frontend/tree/master/src/components/date-field
 *
 */

var DateField = function DateField(_ref2) {
  var children = _ref2.children,
      errorText = _ref2.errorText,
      hintText = _ref2.hintText,
      inputNames = _ref2.inputNames,
      defaultValues = _ref2.defaultValues,
      input = _ref2.input,
      props = _objectWithoutPropertiesLoose(_ref2, ["children", "errorText", "hintText", "inputNames", "defaultValues", "input"]);

  return React.createElement(StyledContainer, _extends({}, props, {
    errorText: errorText
  }), React.createElement(LabelText, {
    errorText: errorText
  }, children), hintText && React.createElement(HintText, null, hintText), errorText && React.createElement(ErrorText, {
    errorText: errorText
  }, errorText), React.createElement(Input, _extends({
    names: inputNames // TODO: defaultValues should be a prop on input
    ,
    defaultValues: defaultValues // TODO: allow each individual input (day, month, year) to have a separate bool for error
    ,
    error: !!errorText
  }, input)));
};

DateField.defaultProps = {
  hintText: undefined,
  errorText: undefined,
  inputNames: {
    day: undefined,
    month: undefined,
    year: undefined
  },
  defaultValues: {
    day: undefined,
    month: undefined,
    year: undefined
  },
  input: undefined
};
export default DateField;
