// @govuk-react/button/es/fixtures.js
import React from 'react';
import { boolean, text } from '@storybook/addon-knobs/react';
import { ButtonArrow } from '@govuk-react/icons';
import { BLUE, TEXT_COLOUR, YELLOW, GREY_3, ORANGE } from 'govuk-colours';
import { MemoryRouter, Route, Link } from 'react-router-dom';
import Button from '.';

var ButtonWithKnobs = function ButtonWithKnobs() {
  return React.createElement(Button, {
    mb: "0",
    disabled: boolean('Disabled', false),
    start: boolean('Start', false)
  }, text('Children', 'Button'));
};

var ButtonStart = function ButtonStart() {
  return React.createElement(Button, {
    start: true
  }, "Save and continue");
};

var ButtonStartIcon = function ButtonStartIcon() {
  return React.createElement(Button, {
    icon: React.createElement(ButtonArrow, null),
    start: true
  }, "Start now");
};

var ButtonDisabled = function ButtonDisabled() {
  return React.createElement(Button, {
    disabled: true
  }, "Disabled primary button");
};

var ButtonDisabledStartIcon = function ButtonDisabledStartIcon() {
  return React.createElement(Button, {
    disabled: true,
    start: true,
    icon: React.createElement(ButtonArrow, null)
  }, "Start now");
};

var ButtonBlue = function ButtonBlue() {
  return React.createElement(Button, {
    buttonColour: BLUE
  }, "Blue button");
};

var ButtonWacky = function ButtonWacky() {
  return React.createElement(Button, {
    buttonColour: GREY_3,
    buttonHoverColour: YELLOW,
    buttonShadowColour: ORANGE,
    buttonTextColour: TEXT_COLOUR
  }, "Wacky colours");
};

var ButtonAsLink = function ButtonAsLink() {
  return React.createElement(MemoryRouter, null, React.createElement("div", null, React.createElement(Button, {
    as: Link,
    to: "/"
  }, "Home Link button"), React.createElement("br", null), React.createElement(Button, {
    as: Link,
    to: "/test"
  }, "Test Link button"), React.createElement("br", null), React.createElement(Route, {
    path: "/",
    exact: true,
    render: function render() {
      return React.createElement("div", null, "Home path");
    }
  }), React.createElement(Route, {
    path: "/test",
    render: function render() {
      return React.createElement("div", null, "Test path");
    }
  })));
};

export default Button;
export { ButtonWithKnobs, ButtonStart, ButtonStartIcon, ButtonDisabled, ButtonDisabledStartIcon, ButtonBlue, ButtonWacky, ButtonAsLink };
