// @govuk-react/label/es/index.js
// https://github.com/alphagov/govuk-frontend/tree/master/src/components/
import styled from 'styled-components';
import React from 'react';
import PropTypes from 'prop-types';
import { ERROR_COLOUR } from 'govuk-colours';
import { SPACING } from '@govuk-react/constants';
import { spacing } from '@govuk-react/lib'; // TODO consider removing this, as it's not as per govuk-frontend
// NB our approach to labels/fields differs at present, which is why we have this
// we have no `form-group` - this, to an extent, replaces it...

var StyledLabel = styled('label')({
  display: 'flex',
  flexDirection: 'column',
  boxSizing: 'border-box',
  ':after': {
    content: "''",
    display: 'table',
    clear: 'both'
  }
}, function (_ref) {
  var error = _ref.error;
  return {
    borderLeft: error ? "4px solid " + ERROR_COLOUR : undefined,
    marginRight: error ? SPACING.SCALE_3 : undefined,
    paddingLeft: error ? SPACING.SCALE_2 : undefined
  };
}, spacing.withWhiteSpace({
  marginBottom: 0
}));
/**
 *
 * ### Usage
 *
 * Simple
 * ```jsx
 * <Label>Example</Label>
 * ```
 *
 * ### References:
 * - https://github.com/alphagov/govuk-frontend/tree/master/src/components/
 *
 */

var Label = function Label(props) {
  return React.createElement(StyledLabel, props);
};

Label.defaultProps = {
  error: false
};
export default Label;
