// @govuk-react/footer/es/atoms/copyright/index.js
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Link from '@govuk-react/link';
import { FOOTER_LINK, FOOTER_LINK_HOVER } from 'govuk-colours';
var StyledCopyright = styled(Link)({
  display: 'inline-block',
  textAlign: 'center',
  textDecoration: 'none',
  whiteSpace: 'nowrap',
  ':link, :visited': {
    color: FOOTER_LINK
  },
  ':hover': {
    color: FOOTER_LINK_HOVER
  }
}, function (_ref) {
  var image = _ref.image;
  return image && {
    minWidth: image.width,
    paddingTop: image.height + 10,
    backgroundImage: "url(" + image.source + ")",
    backgroundRepeat: 'no-repeat',
    backgroundPosition: '50% 0%',
    backgroundSize: image.width + "px " + image.height + "px"
  };
});

var Copyright = function Copyright(_ref2) {
  var text = _ref2.text,
      image = _ref2.image,
      link = _ref2.link;
  return React.createElement(StyledCopyright, {
    href: link,
    image: image
  }, "\xA9 ", text);
};

Copyright.defaultProps = {
  text: undefined,
  image: undefined,
  link: undefined
};
export default Copyright;
