// @govuk-react/details/es/fixtures.js
import React from 'react';
import { boolean, text } from '@storybook/addon-knobs/react';
import Paragraph from '@govuk-react/paragraph';
import Details from '.';
export var summaryText = 'Help with nationality';
export var paragraphs = ['If you’re not sure about your nationality, try to find out from an official document like a passport or national ID card.', 'We need to know your nationality so we can work out which elections you’re entitled to vote in. If you can’t provide your nationality, you’ll have to send copies of identity documents through the post.'];

var DetailsSimple = function DetailsSimple() {
  return React.createElement(Details, {
    summary: summaryText
  }, text('Contents', paragraphs[0]));
};

var DetailsOpen = function DetailsOpen() {
  return React.createElement(Details, {
    summary: summaryText,
    open: true
  }, text('Contents', paragraphs[0]));
};

var DetailsParagraph = function DetailsParagraph() {
  return React.createElement(Details, {
    summary: summaryText,
    open: boolean('Open', false)
  }, React.createElement(Paragraph, null, text('Contents', paragraphs[0] + "\n\n" + paragraphs[1])));
};

export { DetailsSimple, DetailsOpen, DetailsParagraph };
