// @govuk-react/footer/es/fixtures.js
var _styled;

import React from 'react';
import { BrowserRouter, Link, Route } from 'react-router-dom';
import styled from 'styled-components';
import { MEDIA_QUERIES, GUTTER } from '@govuk-react/constants';
import Footer from '.';
import crest from './govuk-crest.png';

var withRoutes = function withRoutes(component) {
  return React.createElement(BrowserRouter, null, component, React.createElement(Route, {
    path: "/footer-nav-item-2",
    render: function render() {
      return 'Footer Navigation item 2';
    }
  }), React.createElement(Route, {
    path: "/footer-meta-item-2",
    render: function render() {
      return 'Footer Meta item 2';
    }
  }));
};

export default (function () {
  return React.createElement(Footer, null);
});
var copyright = {
  text: 'Crown copyright',
  link: 'https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/',
  image: {
    source: crest,
    height: 102,
    width: 125
  }
};
var metaLinks = React.createElement(Footer.MetaLinks, {
  heading: "Support links"
}, React.createElement(Footer.Link, {
  href: "/"
}, "Item 1"), React.createElement(Footer.Link, {
  to: "/footer-meta-item-2",
  as: Link
}, "Item 2 (Router Link)"), React.createElement(Footer.Link, {
  href: "/"
}, "Item 3"));
var metaCustom = React.createElement(Footer.MetaCustom, null, "Built by the ", React.createElement(Footer.Link, {
  href: "/"
}, "Government Digital Service"));
var metaLinksAndCustom = React.createElement(React.Fragment, null, metaLinks, metaCustom);
var navigation = React.createElement(Footer.Navigation, null, React.createElement(Footer.NavigationLinks, {
  heading: "Two column list",
  listColumns: 2
}, React.createElement(Footer.Link, {
  href: "/"
}, "Navigation item 1"), React.createElement(Footer.Link, {
  to: "/footer-nav-item-2",
  as: Link
}, "Navigation item 2 (Router Link)"), React.createElement(Footer.Link, {
  href: "/"
}, "Navigation item 3"), React.createElement(Footer.Link, {
  href: "/"
}, "Navigation item 4"), React.createElement(Footer.Link, {
  href: "/"
}, "Navigation item 5"), React.createElement(Footer.Link, {
  href: "/"
}, "Navigation item 6")), React.createElement(Footer.NavigationLinks, {
  heading: "Single column list"
}, React.createElement(Footer.Link, {
  href: "/"
}, "Navigation item 1"), React.createElement(Footer.Link, {
  href: "/"
}, "Navigation item 2"), React.createElement(Footer.Link, {
  href: "/"
}, "Navigation item 3")));
var WideContainer = styled(Footer.WidthContainer)((_styled = {
  maxWidth: 'inherit'
}, _styled[MEDIA_QUERIES.MAX] = {
  margin: "0 " + GUTTER
}, _styled));
export var FooterWithCustomWidth = function FooterWithCustomWidth() {
  return React.createElement(Footer, {
    container: WideContainer
  });
};
export var FooterWithCopyright = function FooterWithCopyright() {
  return React.createElement(Footer, {
    copyright: copyright
  });
};
export var FooterWithMetaLinks = function FooterWithMetaLinks() {
  return withRoutes(React.createElement(Footer, {
    meta: metaLinks
  }));
};
export var FooterWithSingleMetaLink = function FooterWithSingleMetaLink() {
  return withRoutes(React.createElement(Footer, {
    meta: React.createElement(Footer.MetaLinks, {
      heading: "Support links"
    }, React.createElement(Footer.Link, {
      href: "/"
    }, "Item 1"))
  }));
};
export var FooterWithMetaCustom = function FooterWithMetaCustom() {
  return React.createElement(Footer, {
    meta: metaCustom
  });
};
export var FooterWithMetaLinksAndCustom = function FooterWithMetaLinksAndCustom() {
  return withRoutes(React.createElement(Footer, {
    meta: metaLinksAndCustom
  }));
};
export var FooterWithSingleNavigationLink = function FooterWithSingleNavigationLink() {
  return React.createElement(Footer, null, React.createElement(Footer.Navigation, null, React.createElement(Footer.NavigationLinks, {
    heading: "Single column list"
  }, React.createElement(Footer.Link, {
    href: "/"
  }, "Navigation item 1"))));
};
export var FooterWithNavigation = function FooterWithNavigation() {
  return withRoutes(React.createElement(Footer, null, navigation));
};
export var FooterWithNavigationAndMeta = function FooterWithNavigationAndMeta() {
  return withRoutes(React.createElement(Footer, {
    meta: metaLinksAndCustom
  }, navigation));
};
