// @govuk-react/paragraph/es/fixtures.js
import React from 'react';
import { text, boolean } from '@storybook/addon-knobs/react';
import Paragraph from '.';
var exampleParagraph = 'Paragraph fields support markdown and allow for **bold** text and *italics*, as well as [links](https://en.wikipedia.org/wiki/Markdown). `Inline` code snippets are also supported, as well as code blocks PROVIDED they start on their own line';
var exampleCodeBlock = "```\n  const CustomStyledButton = styled(Button)({\n    backgroundColor: WHITE,\n    borderStyle: 'solid',\n    borderWidth: '2px',\n    borderColor: BLUE,\n    color: BLUE,\n    ':hover': {\n      backgroundColor: BLUE,\n      color: WHITE\n    }\n  });\n  ```";

var ParagraphWithKnobs = function ParagraphWithKnobs() {
  return React.createElement(Paragraph, {
    supportingText: boolean('supportingText', false)
  }, text('children', exampleParagraph));
};

var SupportingParagraph = function SupportingParagraph() {
  return React.createElement(Paragraph, {
    supportingText: true
  }, text('children', exampleParagraph));
};

var ParagraphWithCodeBlock = function ParagraphWithCodeBlock() {
  return React.createElement(Paragraph, {
    supportingText: true
  }, text('children', exampleCodeBlock));
};

export default Paragraph;
export { ParagraphWithKnobs, SupportingParagraph, ParagraphWithCodeBlock, exampleParagraph, exampleCodeBlock };
