// @govuk-react/tabs/es/fixtures.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { BREAKPOINTS } from '@govuk-react/constants';
import { H2, H4 } from '@govuk-react/heading';
import SectionBreak from '@govuk-react/section-break';
import Table from '@govuk-react/table';
import Tabs from '.';

var flip2dArray = function flip2dArray(prev, next) {
  return next.map(function (item, index) {
    return [].concat(prev[index] || [], [next[index]]);
  });
};

function setTabIndex(tabIndex) {
  return this.setState({
    tabIndex: tabIndex
  });
}

function handleClick(_ref) {
  var e = _ref.event,
      index = _ref.index;

  /* eslint-disable-next-line no-undef */
  var mql = window.matchMedia("(min-width: " + BREAKPOINTS.TABLET + ")");

  if (mql.matches) {
    e.preventDefault();
  }

  return this.setTabIndex(index);
}

var sharedDefaultProps = {
  defaultIndex: 0
};
var tableHead = React.createElement(Table.Row, null, React.createElement(Table.CellHeader, null, "Case manager"), React.createElement(Table.CellHeader, null, "Cases opened"), React.createElement(Table.CellHeader, null, "Cases closed"));
var arrTabularTabs = [{
  title: 'Past day',
  arr: [[3, 1, 2], [0, 0, 0]],
  id: 'past-day'
}, {
  title: 'Past week',
  arr: [[24, 16, 24], [18, 20, 27]],
  id: 'past-week'
}, {
  title: 'Past month',
  arr: [[98, 122, 126], [95, 131, 142]],
  id: 'past-month'
}, {
  title: 'Past year',
  arr: [[1380, 1129, 1539], [1472, 1083, 1265]],
  id: 'past-year'
}];

var TableTabs =
/*#__PURE__*/
function (_Component) {
  _inheritsLoose(TableTabs, _Component);

  function TableTabs(props) {
    var _this;

    _this = _Component.call(this, props) || this;
    _this.state = {
      tabIndex: props.defaultIndex
    };
    _this.setTabIndex = setTabIndex.bind(_assertThisInitialized(_this));
    _this.handleClick = handleClick.bind(_assertThisInitialized(_this));
    return _this;
  }

  var _proto = TableTabs.prototype;

  _proto.render = function render() {
    var _this2 = this;

    var tabIndex = this.state.tabIndex;
    return React.createElement(Tabs, null, React.createElement(Tabs.Title, null, "Contents"), React.createElement(Tabs.List, null, arrTabularTabs.map(function (_ref2, index) {
      var id = _ref2.id,
          title = _ref2.title;
      return React.createElement(Tabs.Tab, {
        key: title + "-tabHeader",
        onClick: function onClick(event) {
          return _this2.handleClick({
            event: event,
            index: index
          });
        },
        href: "#" + id,
        selected: tabIndex === index
      }, title);
    })), arrTabularTabs.map(function (_ref3, index) {
      var arr = _ref3.arr,
          id = _ref3.id,
          title = _ref3.title;
      return React.createElement(Tabs.Panel, {
        selected: tabIndex === index,
        key: title + "-tabPanel",
        id: id
      }, React.createElement(H2, null, title), React.createElement(Table, {
        head: tableHead
      }, [['David Francis', 'Paul Farmer', 'Rita Patel']].concat(arr).reduce(flip2dArray, []).map(function (innerArr) {
        return React.createElement(Table.Row, {
          key: innerArr.join() + "-col"
        }, innerArr.map(function (elem) {
          return React.createElement(Table.Cell, {
            key: elem + "-row"
          }, elem);
        }));
      })));
    }));
  };

  return TableTabs;
}(Component);

TableTabs.defaultProps = sharedDefaultProps;

/* eslint-disable-next-line react/no-multi-comp */
var SimpleTabs =
/*#__PURE__*/
function (_Component2) {
  _inheritsLoose(SimpleTabs, _Component2);

  function SimpleTabs(props) {
    var _this3;

    _this3 = _Component2.call(this, props) || this;
    _this3.state = {
      tabIndex: props.defaultIndex
    };
    _this3.setTabIndex = setTabIndex.bind(_assertThisInitialized(_this3));
    _this3.handleClick = handleClick.bind(_assertThisInitialized(_this3));
    return _this3;
  }

  var _proto2 = SimpleTabs.prototype;

  _proto2.render = function render() {
    var _this4 = this;

    var tabIndex = this.state.tabIndex;
    return React.createElement(Tabs, null, React.createElement(Tabs.Title, null, "Contents"), React.createElement(Tabs.List, null, React.createElement(Tabs.Tab, {
      onClick: function onClick(event) {
        return _this4.handleClick({
          event: event,
          index: 0
        });
      },
      selected: tabIndex === 0,
      href: "#first-panel"
    }, "Title 1"), React.createElement(Tabs.Tab, {
      onClick: function onClick(event) {
        return _this4.handleClick({
          event: event,
          index: 1
        });
      },
      selected: tabIndex === 1,
      href: "#second-panel"
    }, "Title 2")), React.createElement(Tabs.Panel, {
      selected: tabIndex === 0,
      id: "first-panel"
    }, "Panel content 1"), React.createElement(Tabs.Panel, {
      selected: tabIndex === 1,
      id: "second-panel"
    }, "Panel content 2"));
  };

  return SimpleTabs;
}(Component);

SimpleTabs.defaultProps = sharedDefaultProps;
var arrSimpleMapped = [{
  contentListItem: 'Title 1',
  contentPanel: 'Mapped Panel content 1',
  id: 'first-panel'
}, {
  contentListItem: 'Title 2',
  contentPanel: 'Mapped Panel content 2',
  id: 'second-panel'
}];
/* eslint-disable-next-line react/no-multi-comp */

var SimpleMapTabs =
/*#__PURE__*/
function (_Component3) {
  _inheritsLoose(SimpleMapTabs, _Component3);

  function SimpleMapTabs(props) {
    var _this5;

    _this5 = _Component3.call(this, props) || this;
    _this5.state = {
      tabIndex: props.defaultIndex
    };
    _this5.setTabIndex = setTabIndex.bind(_assertThisInitialized(_this5));
    _this5.handleClick = handleClick.bind(_assertThisInitialized(_this5));
    return _this5;
  }

  var _proto3 = SimpleMapTabs.prototype;

  _proto3.render = function render() {
    var _this6 = this;

    var tabIndex = this.state.tabIndex;
    return React.createElement(Tabs, null, React.createElement(Tabs.Title, null, "Contents"), React.createElement(Tabs.List, null, arrSimpleMapped.map(function (_ref4, index) {
      var contentListItem = _ref4.contentListItem,
          id = _ref4.id;
      return React.createElement(Tabs.Tab, {
        key: contentListItem + "-simpleMappedListItem",
        onClick: function onClick(event) {
          return _this6.handleClick({
            event: event,
            index: index
          });
        },
        selected: tabIndex === index,
        href: "#" + id
      }, contentListItem);
    })), arrSimpleMapped.map(function (_ref5, index) {
      var contentPanel = _ref5.contentPanel,
          id = _ref5.id;
      return React.createElement(Tabs.Panel, {
        key: contentPanel + "-simpleMappedPanel",
        selected: tabIndex === index,
        id: id
      }, React.createElement(H4, null, contentPanel));
    }));
  };

  return SimpleMapTabs;
}(Component);

SimpleMapTabs.defaultProps = sharedDefaultProps;
var arrProposedBabel = [{
  contentListItem: 'Title 1',
  contentPanel: 'Panel content 1',
  id: 'first-panel'
}, {
  contentListItem: 'Title 2',
  contentPanel: 'Panel content 2',
  id: 'second-panel'
}];
/* eslint-disable-next-line react/no-multi-comp */

var ProposedClassPropertiesPlugin =
/*#__PURE__*/
function (_Component4) {
  _inheritsLoose(ProposedClassPropertiesPlugin, _Component4);

  function ProposedClassPropertiesPlugin() {
    var _this7;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this7 = _Component4.call.apply(_Component4, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this7), "state", {
      tabIndex: _this7.props.defaultIndex
    });

    _defineProperty(_assertThisInitialized(_this7), "setTabIndex", setTabIndex);

    _defineProperty(_assertThisInitialized(_this7), "handleClick", handleClick);

    return _this7;
  }

  var _proto4 = ProposedClassPropertiesPlugin.prototype;

  _proto4.render = function render() {
    var _this8 = this;

    var tabIndex = this.state.tabIndex;
    return React.createElement(Tabs, null, React.createElement(Tabs.Title, null, "Contents"), React.createElement(Tabs.List, null, arrProposedBabel.map(function (_ref6, index) {
      var contentListItem = _ref6.contentListItem,
          id = _ref6.id;
      return React.createElement(Tabs.Tab, {
        key: contentListItem + "-babelListItem",
        onClick: function onClick(event) {
          return _this8.handleClick({
            event: event,
            index: index
          });
        },
        selected: tabIndex === index,
        href: "#" + id
      }, contentListItem);
    })), arrProposedBabel.map(function (_ref7, index) {
      var contentPanel = _ref7.contentPanel,
          id = _ref7.id;
      return React.createElement(Tabs.Panel, {
        key: contentPanel + "-babelPanel",
        selected: tabIndex === index,
        id: id
      }, React.createElement(H2, null, "with Babel Plugin"), React.createElement(SectionBreak, {
        level: "L",
        visible: true
      }), contentPanel);
    }));
  };

  return ProposedClassPropertiesPlugin;
}(Component);

_defineProperty(ProposedClassPropertiesPlugin, "defaultProps", sharedDefaultProps);

var HooksExample = function HooksExample(_ref8) {
  var defaultIndex = _ref8.defaultIndex;

  var _React$useState = React.useState(defaultIndex),
      tabIndex = _React$useState[0],
      setHooksTabIndex = _React$useState[1];

  var handleTabChange = function handleTabChange(newTabIndex) {
    return setHooksTabIndex(newTabIndex);
  };

  function hooksHandleClick(_ref9) {
    var e = _ref9.event,
        index = _ref9.index;

    /* eslint-disable-next-line no-undef */
    var mql = window.matchMedia("(min-width: " + BREAKPOINTS.TABLET + ")");

    if (mql.matches) {
      e.preventDefault();
    }

    return handleTabChange(index);
  }

  return React.createElement(Tabs, null, React.createElement(Tabs.Title, null, "Content"), React.createElement(Tabs.List, null, [{
    content: 'Hooks Title 1',
    href: '#first-panel'
  }, {
    content: 'Hooks Title 2',
    href: '#second-panel'
  }].map(function (_ref10, index) {
    var content = _ref10.content,
        href = _ref10.href;
    return React.createElement(Tabs.Tab, {
      onClick: function onClick(event) {
        return hooksHandleClick({
          event: event,
          index: index
        });
      },
      selected: tabIndex === index,
      href: href
    }, content);
  })), [{
    content: 'Hooks Panel content 1',
    id: 'first-panel'
  }, {
    content: 'Hooks Panel content 2',
    id: 'second-panel'
  }].map(function (_ref11, index) {
    var content = _ref11.content,
        id = _ref11.id;
    return React.createElement(Tabs.Panel, {
      selected: tabIndex === index,
      id: id
    }, content);
  }));
};

HooksExample.defaultProps = sharedDefaultProps;
export { HooksExample, ProposedClassPropertiesPlugin, SimpleTabs, SimpleMapTabs, TableTabs };
