// @govuk-react/link/es/fixtures.js
import React from 'react';
import { boolean, text } from '@storybook/addon-knobs/react';
import { BrowserRouter, Link as RLink, Route } from 'react-router-dom';
import Link, { LinkDocumented } from '.';
export default (function () {
  return (// eslint-disable-next-line jsx-a11y/anchor-is-valid
    React.createElement(LinkDocumented, {
      muted: boolean('muted', false),
      textColour: boolean('textColour', false),
      noVisitedState: boolean('noVisitedState', false),
      href: "#"
    }, text('Link content', 'Link example'))
  );
});
export var LinkWithAllOptionsOn = function LinkWithAllOptionsOn() {
  return (// eslint-disable-next-line jsx-a11y/anchor-is-valid
    React.createElement(LinkDocumented, {
      muted: true,
      textColour: true,
      noVisitedState: true,
      href: "#"
    }, text('Link content', 'Link example'))
  );
};
export var LinkWithRouter = function LinkWithRouter() {
  return React.createElement(BrowserRouter, null, React.createElement("div", null, React.createElement(Link, {
    as: RLink,
    to: "/first"
  }, "First link"), React.createElement("br", null), React.createElement(Link, {
    as: RLink,
    to: "/second"
  }, "Second link"), React.createElement("br", null), React.createElement("div", null, React.createElement(Route, {
    path: "/first",
    render: function render() {
      return 'First route';
    }
  }), React.createElement(Route, {
    path: "/second",
    render: function render() {
      return 'Second route';
    }
  }))));
};
