// @govuk-react/table/es/fixtures.js
import React from 'react';
import Table from '.';

var TableSimple = function TableSimple() {
  return React.createElement(Table, null, React.createElement(Table.Row, null, React.createElement(Table.CellHeader, null, "First 6 weeks"), React.createElement(Table.Cell, null, "\xA3109.80 per week")), React.createElement(Table.Row, null, React.createElement(Table.CellHeader, null, "Next 33 weeks"), React.createElement(Table.Cell, null, "\xA3109.80 per week")), React.createElement(Table.Row, null, React.createElement(Table.CellHeader, null, "Total estimated pay"), React.createElement(Table.Cell, null, "\xA34,282.20")), React.createElement(Table.Row, null, React.createElement(Table.CellHeader, null, "Tell the mother\u2019s employer"), React.createElement(Table.Cell, null, "28 days before they want to start maternity pay")));
};

var TableWithCaption = function TableWithCaption() {
  return React.createElement(Table, {
    caption: "Dates and amounts",
    body: React.createElement(React.Fragment, null, React.createElement(Table.Row, null, React.createElement(Table.CellHeader, null, "First 6 weeks"), React.createElement(Table.Cell, null, "\xA3109.80 per week")), React.createElement(Table.Row, null, React.createElement(Table.CellHeader, null, "Next 33 weeks"), React.createElement(Table.Cell, null, "\xA3109.80 per week")), React.createElement(Table.Row, null, React.createElement(Table.CellHeader, null, "Total estimated pay"), React.createElement(Table.Cell, null, "\xA34,282.20")))
  });
};

var exampleHead = React.createElement(Table.Row, null, React.createElement(Table.CellHeader, null, "Month you apply"), React.createElement(Table.CellHeader, {
  numeric: true
}, "Rate for vehicles"), React.createElement(Table.CellHeader, {
  numeric: true
}, "Rate for bicycles"));

var TableWithHeadAndNumerics = function TableWithHeadAndNumerics() {
  return React.createElement(Table, {
    caption: "Attention, I am the caption of this ship!",
    head: exampleHead
  }, React.createElement(Table.Row, null, React.createElement(Table.CellHeader, null, "January"), React.createElement(Table.Cell, {
    numeric: true
  }, "\xA3165.00"), React.createElement(Table.Cell, {
    numeric: true
  }, "\xA385.00")), React.createElement(Table.Row, null, React.createElement(Table.CellHeader, null, "February"), React.createElement(Table.Cell, {
    numeric: true
  }, "\xA3165.00"), React.createElement(Table.Cell, {
    numeric: true
  }, "\xA385.00")), React.createElement(Table.Row, null, React.createElement(Table.CellHeader, null, "March"), React.createElement(Table.Cell, {
    numeric: true
  }, "\xA3151.00"), React.createElement(Table.Cell, {
    numeric: true
  }, "\xA377.00")), React.createElement(Table.Row, null, React.createElement(Table.CellHeader, null, "April"), React.createElement(Table.Cell, {
    numeric: true
  }, "\xA3136.00"), React.createElement(Table.Cell, {
    numeric: true
  }, "\xA370.00")));
};

var TableThreeQuartersOneQuarter = function TableThreeQuartersOneQuarter() {
  return React.createElement(Table, {
    caption: "Three quarters, one quarter",
    head: React.createElement(Table.Row, null, React.createElement(Table.CellHeader, {
      setWidth: "three-quarters"
    }, "three-quarters"), React.createElement(Table.CellHeader, {
      setWidth: "one-quarter"
    }, "one-quarter"))
  }, React.createElement(Table.Row, null, React.createElement(Table.Cell, null, "Column uses setWidth=\"three-quarters\" in header"), React.createElement(Table.Cell, null, "setWidth=\"one-quarter\"")));
};

var TableWithCustomWidths = function TableWithCustomWidths() {
  return React.createElement(Table, {
    caption: "Custom widths",
    head: React.createElement(Table.Row, null, React.createElement(Table.CellHeader, {
      setWidth: "one-half"
    }, "one-half"), React.createElement(Table.CellHeader, {
      setWidth: "22%"
    }, "22%"), React.createElement(Table.CellHeader, null, "Normal"))
  }, React.createElement(Table.Row, null, React.createElement(Table.Cell, null, "Column uses setWidth=\"one-half\" in header"), React.createElement(Table.Cell, null, "setWidth=\"22%\""), React.createElement(Table.Cell, null, "Not specified")));
};

export { TableSimple, TableWithCaption, TableWithHeadAndNumerics, TableWithCustomWidths, TableThreeQuartersOneQuarter };
export default Table;
