// @govuk-react/error-summary/es/fixtures.js
/* eslint-env browser */
import React from 'react';
import { text } from '@storybook/addon-knobs/react';
import InputField from '@govuk-react/input-field';
import TextArea from '@govuk-react/text-area';
import ErrorSummary from '.';
export var heading = 'Message to alert the user to a problem goes here';
export var description = 'Optional description of the errors and how to correct them';
export var errors = [{
  targetName: 'national-insurance-number',
  text: 'National Insurance number error'
}, {
  targetName: 'description',
  text: 'Description of what you saw error'
}];

var onHandleErrorClick = function onHandleErrorClick(targetName) {
  document.getElementsByName(targetName)[0].scrollIntoView();
};

export default (function () {
  return React.createElement("div", null, React.createElement(ErrorSummary, {
    heading: text('heading', heading),
    description: text('description', description),
    onHandleErrorClick: onHandleErrorClick,
    errors: errors
  }), React.createElement(InputField, {
    name: "national-insurance-number",
    hint: "It\u2019s on your National Insurance card, benefit letter, payslip or P60. For example, \u2018QQ 12 34 56 C\u2019."
  }, "National Insurance number"), React.createElement("br", null), React.createElement(TextArea, {
    name: "description"
  }, "Description of what you saw"));
});
