// @govuk-react/heading/es/presets.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { LEVEL_SIZE } from '@govuk-react/constants';
import Heading from '.';
export var H1 = function H1(props) {
  return React.createElement(Heading, _extends({
    as: "h1",
    size: LEVEL_SIZE[1]
  }, props));
};
export var H2 = function H2(props) {
  return React.createElement(Heading, _extends({
    as: "h2",
    size: LEVEL_SIZE[2]
  }, props));
};
export var H3 = function H3(props) {
  return React.createElement(Heading, _extends({
    as: "h3",
    size: LEVEL_SIZE[3]
  }, props));
};
export var H4 = function H4(props) {
  return React.createElement(Heading, _extends({
    as: "h4",
    size: LEVEL_SIZE[4]
  }, props));
};
export var H5 = function H5(props) {
  return React.createElement(Heading, _extends({
    as: "h5",
    size: LEVEL_SIZE[5]
  }, props));
};
export var H6 = function H6(props) {
  return React.createElement(Heading, _extends({
    as: "h6",
    size: LEVEL_SIZE[6]
  }, props));
};
