// @govuk-react/phase-banner/es/index.js
var _styled;

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { BORDER_COLOUR } from 'govuk-colours';
import { SPACING_POINTS } from '@govuk-react/constants';
import { spacing, typography } from '@govuk-react/lib';
import Tag from '@govuk-react/tag';
var StyledBanner = styled('div')({
  paddingTop: SPACING_POINTS[2],
  paddingBottom: SPACING_POINTS[2],
  borderBottom: "1px solid " + BORDER_COLOUR
}, spacing.withWhiteSpace());
var BannerContent = styled('p')(typography.font({
  size: 16
}), typography.textColour, (_styled = {
  display: 'table',
  margin: 0
}, _styled[Tag] = {
  marginRight: SPACING_POINTS[2]
}, _styled));
var BannerText = styled('span')({
  display: 'table-cell',
  verticalAlign: 'baseline'
});
/**
 *
 * ### Usage
 *
 * Alpha
 * ```jsx
 * <PhaseBanner level="alpha">
 *    This part of GOV.UK is being rebuilt &#8211;{' '}
 *    <AnchorLink href="https://example.com">find out what that means</AnchorLink>
 *  </PhaseBanner>
 * ```
 *
 * Beta
 * ```jsx
 * <PhaseBanner level="beta">
 *    This part of GOV.UK is being rebuilt &#8211;{' '}
 *    <AnchorLink href="https://example.com">find out what that means</AnchorLink>
 *  </PhaseBanner>
 * ```
 *
 * ### References:
 * - https://github.com/alphagov/govuk-frontend/tree/master/src/components/phase-banner
 * - https://design-system.service.gov.uk/components/phase-banner/
 *
 */

var PhaseBanner = function PhaseBanner(_ref) {
  var level = _ref.level,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["level", "children"]);

  return React.createElement(StyledBanner, props, React.createElement(BannerContent, null, React.createElement(Tag, null, level), React.createElement(BannerText, null, children)));
};

export default PhaseBanner;
