// @govuk-react/multi-choice/es/fixtures.js
import React, { Fragment } from 'react';
import { text, object } from '@storybook/addon-knobs/react';
import Radio from '@govuk-react/radio';
import MultiChoice from '.';
var metaData = {
  touched: true,
  error: 'Example error'
};
var children = React.createElement(Fragment, null, React.createElement(Radio, {
  name: "group1",
  inline: true
}, "Yes"), React.createElement(Radio, {
  name: "group1",
  inline: true
}, "No"));

var MultiChoiceWithKnobs = function MultiChoiceWithKnobs() {
  return React.createElement(MultiChoice, {
    label: text('Label', 'Example label')
  }, children);
};

export var MultiChoiceWithKnobsHint = function MultiChoiceWithKnobsHint() {
  return React.createElement(MultiChoice, {
    label: text('Label', 'Example label'),
    hint: text('Example hint', 'Example hint')
  }, children);
};
export var MultiChoiceWithKnobsError = function MultiChoiceWithKnobsError() {
  return React.createElement(MultiChoice, {
    label: text('Label', 'Example label'),
    meta: object('Meta', metaData)
  }, children);
};
export default MultiChoiceWithKnobs;
