// @govuk-react/document-footer-metadata/es/index.js
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import UnorderedList from '@govuk-react/unordered-list';
import { NTA_LIGHT } from '@govuk-react/constants';
import { spacing } from '@govuk-react/lib';
var StyledContainer = styled('div')({
  fontFamily: NTA_LIGHT
}, spacing.withWhiteSpace());
var StyledDefinition = styled('li')({
  fontSize: '24px',
  fontWeight: 700,
  lineHeight: 1.25,
  '> a': {
    textDecoration: 'none'
  }
});
/**
 *
 * ### Usage
 *
 *
 * Simple
 * ```jsx
 * import { Link } from '@govuk-react/link';
 *
 * const fromData = [
 *   <Link href="/government/organisations/ministry-of-defence">
 *     Ministry of Defence
 *   </Link>,
 * ];
 *
 * <DocumentFooterMetadata from={fromData} />
 * ```
 *
 *
 * DFM From & part of example
 * ```jsx
 * import { Link } from '@govuk-react/link';
 *
 * const fromData = [
 *   <Link href="/government/organisations/ministry-of-defence">
 *     Ministry of Defence
 *   </Link>,
 * ];
 * const partOfData = [
 *   <Link href="/government/topics/energy">Energy</Link>,
 *   <Link href="/government/topics/environment">Environment</Link>,
 * ];
 *
 * <DocumentFooterMetadata from={fromData} partOf={partOfData} />
 * ```
 *
 *
 * DFM From & other data example
 * ```jsx
 * import { Link } from '@govuk-react/link';
 *
 * const fromData = [
 *   <Link href="/government/organisations/ministry-of-defence">
 *     Ministry of Defence
 *   </Link>,
 * ];
 * const otherData = [
 *   {
 *     id: 0,
 *     title: 'Consultation type',
 *     content: <Link href="/government/publications">Open</Link>,
 *   },
 *   {
 *     id: 1,
 *     title: 'Published',
 *     content: '20 January 2012',
 *   },
 * ];
 *
 * <DocumentFooterMetadata from={fromData} other={otherData} />
 * ```
 *
 * ### References:
 * - https://govuk-static.herokuapp.com/component-guide/document_footer
 */

var DocumentFooterMetadata = function DocumentFooterMetadata(_ref) {
  var from = _ref.from,
      partOf = _ref.partOf,
      other = _ref.other,
      props = _objectWithoutPropertiesLoose(_ref, ["from", "partOf", "other"]);

  var fromData = React.createElement(StyledContainer, null, from && React.createElement("div", null, React.createElement("p", {
    style: {
      marginBottom: 0
    }
  }, "From:"), React.createElement(UnorderedList, {
    listStyleType: "none"
  }, from && from.map(function (child, i) {
    return (
      /* eslint-disable react/no-array-index-key */
      React.createElement(StyledDefinition, {
        key: i
      }, child)
      /* eslint-enable react/no-array-index-key */

    );
  }))));
  var partOfData = React.createElement(StyledContainer, null, partOf && React.createElement("div", null, React.createElement("p", {
    style: {
      marginBottom: 0
    }
  }, "Part of:"), React.createElement(UnorderedList, {
    listStyleType: "none"
  }, partOf && partOf.map(function (child, i) {
    return React.createElement(StyledDefinition, {
      key: child.key || i
    }, child);
  }))));
  var otherData = React.createElement(StyledContainer, null, other && other.map(function (item) {
    return React.createElement("div", {
      key: item.id
    }, React.createElement("p", {
      style: {
        marginBottom: 0
      }
    }, item.title, ":"), React.createElement(UnorderedList, {
      listStyleType: "none"
    }, React.createElement(StyledDefinition, null, item.content)));
  }));
  return React.createElement(StyledContainer, props, fromData, partOfData, otherData);
};

DocumentFooterMetadata.defaultProps = {
  from: undefined,
  partOf: undefined,
  other: undefined
};
export default DocumentFooterMetadata;
