// @govuk-react/page/es/index.js
// Tracking:
// https://design-system.service.gov.uk/styles/page-template/
// https://design-system.service.gov.uk/styles/layout/#page-wrappers
// https://github.com/alphagov/govuk-frontend/blob/master/src/objects/_main-wrapper.scss
// https://github.com/alphagov/govuk-frontend/blob/master/src/objects/_width-container.scss
import React from 'react';
import PropTypes from 'prop-types';
import SkipLink from '@govuk-react/skip-link';
import TopNav from '@govuk-react/top-nav'; // import Footer from '@govuk-react/footer';

import Main from './atoms/main';
import WidthContainer from './atoms/width-container';
/**
 *
 * ### Usage
 *
 * Simple
 * ```jsx
 * import BackLink from '@govuk-react/back-link';
 * import { H1 } from '@govuk-react/heading';
 *
 * <Page beforeChildren={<BackLink href="#" />}>
 *   <H1>Page title</H1>
 * </Page>
 * ```
 *
 * ### References
 * - https://design-system.service.gov.uk/styles/page-template/
 * - https://design-system.service.gov.uk/styles/layout/#page-wrappers
 * - https://github.com/alphagov/govuk-frontend/blob/master/src/objects/_main-wrapper.scss
 * - https://github.com/alphagov/govuk-frontend/blob/master/src/objects/_width-container.scss
 */

var Page = function Page(_ref) {
  var header = _ref.header,
      footer = _ref.footer,
      children = _ref.children,
      id = _ref.id,
      beforeChildren = _ref.beforeChildren,
      MainComponent = _ref.main,
      Container = _ref.container;
  return React.createElement(React.Fragment, null, React.createElement(SkipLink, {
    href: "#" + id
  }), header, React.createElement(Container, null, beforeChildren, React.createElement(MainComponent, {
    id: id
  }, children)), footer);
};

Page.Main = Main;
Page.WidthContainer = WidthContainer;
Page.defaultProps = {
  children: undefined,
  id: 'content',
  header: React.createElement(TopNav, null),
  footer: undefined,
  // <Footer />, // TODO: add Footer component once built
  main: Page.Main,
  beforeChildren: undefined,
  container: Page.WidthContainer
};
export default Page;
